### DISCUSSION TITLE
Is there a recommended way to test api routes?

### DISCUSSION THREAD

**Discussion by insanity54** [blitz-js/blitz#2159] [Q&A]
I have a Blitz app with some custom logic that is handled by an api route, e.g. `/api/foo`

Is there a recommended way to create unit tests for this api route? In other words, is there a way to test that `/api/foo` responds with a specific value?

**Comment by flybayer** [comment]
I recommend extracting the business logic into a separate function that allows you to test it without mocking `req` and `res` objects. It can be in the same file, exported as a named export.

**Comment by m5r** [comment]
You can also test your api routes with https://github.com/Xunnamius/next-test-api-route-handler
